incubator/hnc/internal/validators/hierarchy.go

```diff
@@ -176,6 +176,11 @@ func (v *Hierarchy) checkNS(ns *forest.Namespace) admission.Response {
 
 // checkParent validates if the parent is legal based on the current in-memory state of the forest.
 func (v *Hierarchy) checkParent(ns, curParent, newParent *forest.Namespace) admission.Response {
+	if ns.IsExternal() && newParent != nil {
+		msg := fmt.Sprintf("Namespace %q is managed by %q that cannot have parent.", ns.Name(), ns.Manager())
```

Comment:
Maybe: "Namespace %q is managed by %q, not HNC, so it cannot have a parent in HNC."